nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/test/java/org/apache/nifi/processors/aws/ml/polly/GetAwsPollyStatusTest.java

```diff
@@ -0,0 +1,124 @@
+                return mockPollyClient;
+            }
+        };
+        runner = TestRunners.newTestRunner(mockGetAwsPollyStatus);
+        runner.addControllerService("awsCredetialProvider", mockAwsCredentialsProvider);
+        runner.enableControllerService(mockAwsCredentialsProvider);
+        runner.setProperty(AWS_CREDENTIALS_PROVIDER_SERVICE, "awsCredetialProvider");
+    }
+
+    @Test
+    public void testPollyTaskInProgress() {
+        ArgumentCaptor<GetSpeechSynthesisTaskRequest> requestCaptor = ArgumentCaptor.forClass(GetSpeechSynthesisTaskRequest.class);
+        GetSpeechSynthesisTaskResult taskResult = new GetSpeechSynthesisTaskResult();
+        SynthesisTask task = new SynthesisTask().withTaskId(TEST_TASK_ID)
+                .withTaskStatus(TaskStatus.InProgress);
+        taskResult.setSynthesisTask(task);
+        when(mockPollyClient.getSpeechSynthesisTask(requestCaptor.capture())).thenReturn(taskResult);
+        runner.enqueue("content", Collections.singletonMap(AWS_TASK_ID_PROPERTY, TEST_TASK_ID));
+        runner.run();
+
+        runner.assertAllFlowFilesTransferred(REL_RUNNING);
+        assertEquals(requestCaptor.getValue().getTaskId(), TEST_TASK_ID);
+    }
+
+    @Test
+    public void testPollyTaskCompleted() {
+        ArgumentCaptor<GetSpeechSynthesisTaskRequest> requestCaptor = ArgumentCaptor.forClass(GetSpeechSynthesisTaskRequest.class);
+        GetSpeechSynthesisTaskResult taskResult = new GetSpeechSynthesisTaskResult();
+        SynthesisTask task = new SynthesisTask().withTaskId(TEST_TASK_ID)
+                .withTaskStatus(TaskStatus.Completed)
+                .withOutputUri("outputLocationPath");
+        taskResult.setSynthesisTask(task);
+        when(mockPollyClient.getSpeechSynthesisTask(requestCaptor.capture())).thenReturn(taskResult);
+        runner.enqueue("content", Collections.singletonMap(AWS_TASK_ID_PROPERTY, TEST_TASK_ID));
+        runner.run();
+
+        runner.assertTransferCount(REL_SUCCESS, 1);
+        runner.assertTransferCount(REL_ORIGINAL, 1);
+        runner.assertAllFlowFilesContainAttribute(REL_SUCCESS, AWS_TASK_OUTPUT_LOCATION);
+        assertEquals(requestCaptor.getValue().getTaskId(), TEST_TASK_ID);
+    }
+
+
+    @Test
+    public void testPollyTaskFailed() {
+        ArgumentCaptor<GetSpeechSynthesisTaskRequest> requestCaptor = ArgumentCaptor.forClass(GetSpeechSynthesisTaskRequest.class);
+        GetSpeechSynthesisTaskResult taskResult = new GetSpeechSynthesisTaskResult();
+        SynthesisTask task = new SynthesisTask().withTaskId(TEST_TASK_ID)
+                .withTaskStatus(TaskStatus.Failed)
+                .withTaskStatusReason("reasonOfFailure");
+        taskResult.setSynthesisTask(task);
+        when(mockPollyClient.getSpeechSynthesisTask(requestCaptor.capture())).thenReturn(taskResult);
+        runner.enqueue("content", Collections.singletonMap(AWS_TASK_ID_PROPERTY, TEST_TASK_ID));
```

Comment:
Recommend defining and reusing a static variable the placeholder content string.
```suggestion
        runner.enqueue(PLACEHOLDER_CONTENT, Collections.singletonMap(AWS_TASK_ID_PROPERTY, TEST_TASK_ID));
```

Reply:
Fixed, thx.